dbms/src/Storages/KVStore/Region.cpp

```diff
@@ -408,34 +415,63 @@ void Region::setRegionTableCtx(RegionTableCtxPtr ctx) const
 
 void Region::maybeWarnMemoryLimitByTable(TMTContext & tmt, const char * from)
```

Comment:
The difference:

1. Get dynamic reloadable limit from `max_memory_usage_for_all_queries` instead of adding a new config copy in `KVStore`
2. If `max_memory_usage_for_all_queries == 0`, then use the server memory capacity * 80%
3. Lower the table level memory threshold from `50% * current` to `min(20% * current, 10GiB)` to increase the recall rate for logging down info. The lowest recommanded memory on official doc is 64GiB. 64GiB * 80% * 20%~10GiB